### PR TITLE
Worker/Jobs.pm: Prevent use of uninitialized value

### PR DESCRIPTION
**Pull request by Martchus**
Show more helpful error message instead

### CODE REVIEW

File: lib/OpenQA/Worker/Jobs.pm [hunk 1]
```diff
@@ -530,7 +530,12 @@ sub upload_status(;$) {
         $status->{log}             = log_snippet("$pooldir/autoinst-log.txt",   \$log_offset);
         $status->{serial_log}      = log_snippet("$pooldir/serial0",            \$serial_offset);
         $status->{serial_terminal} = log_snippet("$pooldir/virtio_console.log", \$serial_terminal_offset);
-        print STDERR $status->{serial_terminal}->{data};
+        if ($status->{serial_terminal}->{data}) {
+            print STDERR $status->{serial_terminal}->{data};
+        }
+        else {
+            print STDERR "Unable to find $pooldir/virtio_console.log\n";
+        }
```
Comment (coolo) [inline lib/OpenQA/Worker/Jobs.pm]: This looks like debug output anyway, just remove it - @richiejp

Comment (Martchus) [inline lib/OpenQA/Worker/Jobs.pm]: I'm not sure whether it is an error when the file is not present. If not, then it would make sense to ignore it.

Comment (coolo) [inline lib/OpenQA/Worker/Jobs.pm]: the file is only present for not-yet existant ltp tests

Comment (Martchus) [inline lib/OpenQA/Worker/Jobs.pm]: Ok, removed it.

**Comment by okurz** [comment]
LGTM

File: lib/OpenQA/Worker/Jobs.pm [hunk 2]
```diff
@@ -530,7 +530,9 @@ sub upload_status(;$) {
         $status->{log}             = log_snippet("$pooldir/autoinst-log.txt",   \$log_offset);
         $status->{serial_log}      = log_snippet("$pooldir/serial0",            \$serial_offset);
         $status->{serial_terminal} = log_snippet("$pooldir/virtio_console.log", \$serial_terminal_offset);
-        print STDERR $status->{serial_terminal}->{data};
+        if ($status->{serial_terminal}->{data}) {
+            print STDERR $status->{serial_terminal}->{data};
```
Comment (richiejp) [inline lib/OpenQA/Worker/Jobs.pm]: I really don't think we want the contents of data to be printed to STDERR except for debugging maybe.

**Comment by richiejp** [comment]
Which value is uninitialized?

File: lib/OpenQA/Worker/Jobs.pm (hunk 2 above)
Comment (Martchus) [inline lib/OpenQA/Worker/Jobs.pm]: So remove it completely?

**Comment by Martchus** [comment]
`$status->{serial_terminal}->{data}` is uninitialized.

**Comment by Martchus** [comment]
Ok, so I removed it completely.

**Comment by richiejp** [comment]
It's just hidden because it was against a particular line which you have removed :-)

**Comment by coolo** [comment]
increasing coverage is always good :)

**Pull request merged**